util/rules.go

```diff
@@ -144,9 +175,9 @@ func (rule Rule) ToGraphiteName(taggedMetric api.TaggedMetric) (GraphiteMetric,
 
 // MatchRule sees if a given graphite string matches
 // any of the specified rules.
-func (ruleSet RuleSet) MatchRule(input string) (api.TaggedMetric, bool) {
-	for _, rule := range ruleSet.rules {
-		value, matched := rule.MatchRule(input)
+func (ruleSet *RuleSet) MatchRule(input string) (api.TaggedMetric, bool) {
+	for i := 0; i < len(ruleSet.Rules); i++ {
```

Comment:
Because I want to modify the underlying struct on a successful match downstream and make sure it stays in memory. Range has weird copy semantics http://stackoverflow.com/questions/15945030/change-values-while-iterating-in-golang . Unless I'm missing something.